stingray/varenergyspectrum.py

```diff
@@ -210,23 +215,28 @@ def _spectrum_function(self):
         lag_spec_err = np.zeros_like(lag_spec)
         for i, eint in enumerate(self.energy_intervals):
             base_lc, ref_lc = self._construct_lightcurves(eint)
-            xspect = AveragedCrossspectrum(base_lc, ref_lc,
-                                           segment_size=self.segment_size)
-            good = (xspect.freq >= self.freq_interval[0]) & \
-                   (xspect.freq < self.freq_interval[1])
-            lag, lag_err = xspect.time_lag()
-            good_lag, good_lag_err = lag[good], lag_err[good]
-            coh, coh_err = xspect.coherence()
-            lag_spec[i] = np.mean(good_lag)
-            coh_check = coh > 1.2 / (1 + 0.2 * xspect.m)
-            if not np.all(coh_check[good]):
-                simon("Coherence is not ideal over the specified energy range."
-                      " Lag values and uncertainties might be underestimated. "
-                      "See Epitropakis and Papadakis, A\&A 591, 1113, 2016")
-
-            # Root squared sum of errors of the spectrum
-            # Verified!
-            lag_spec_err[i] = np.sqrt(np.sum(good_lag_err**2) / len(good_lag))
+            try:
+                xspect = AveragedCrossspectrum(base_lc, ref_lc,
+                                               segment_size=self.segment_size)
+                good = (xspect.freq >= self.freq_interval[0]) & \
+                       (xspect.freq < self.freq_interval[1])
+                lag, lag_err = xspect.time_lag()
+                good_lag, good_lag_err = lag[good], lag_err[good]
+                coh, coh_err = xspect.coherence()
+                lag_spec[i] = np.mean(good_lag)
+                coh_check = coh > 1.2 / (1 + 0.2 * xspect.m)
+                if not np.all(coh_check[good]):
+                    simon("Coherence is not ideal over the specified energy range."
+                          " Lag values and uncertainties might be underestimated. "
+                          "See Epitropakis and Papadakis, A\&A 591, 1113, 2016")
+
+                # Root squared sum of errors of the spectrum
+                # Verified!
+                lag_spec_err[i] = np.sqrt(np.sum(good_lag_err**2) / len(good_lag))
+
+            except AssertionError as e:
+                # Avoid "Mean count rate is <= 0. Something went wrong" assertion.
+                simon("AssertionError: " + str(e))
```

Comment:
Thanks for this!
Maybe I'm missing it, but I can't see the assertion that you're catching! I think it'd be useful for this piece of code to be covered by 1-2 tests. It'd also be great if the `simon` call in the `except` statement could be more verbose about what assertion is failing and why. If I were the user, getting a warning that just reads "AssertionError" wouldn't be very helpful to figuring out what went wrong.